jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/IntermediateEventTest.java

```diff
@@ -2601,4 +2601,44 @@ public void testMultipleTimerBoundaryEventCronCycleVariable() throws Exception {
 
         assertProcessInstanceFinished(processInstance, ksession);
     }
+    
+    @Test(timeout=10000)
+    public void testEventBasedSplitWithCronTimerAndSignal() throws Exception {
+        System.setProperty("jbpm.enable.multi.con", "true");
+        try {
+            CountDownProcessEventListener countDownListener = new CountDownProcessEventListener("Request photos of order in use", 1);
+            CountDownProcessEventListener countDownListener2 = new CountDownProcessEventListener("Request an online review", 1);
+            CountDownProcessEventListener countDownListener3 = new CountDownProcessEventListener("Send a thank you card", 1);
+            CountDownProcessEventListener countDownListener4 = new CountDownProcessEventListener("Request an online review", 1);
+            KieBase kbase = createKnowledgeBase("timer/BPMN2-CronTimerWithEventBasedGateway.bpmn2");
+            ksession = createKnowledgeSession(kbase);
+            
+            TestWorkItemHandler handler = new TestWorkItemHandler();
+            ksession.getWorkItemManager().registerWorkItemHandler("Human Task", handler);       
+            ksession.addEventListener(countDownListener);
+            ksession.addEventListener(countDownListener2);
+            ksession.addEventListener(countDownListener3);
+            ksession.addEventListener(countDownListener4);
+            
+            ProcessInstance processInstance = ksession.startProcess("timerWithEventBasedGateway");
+            assertProcessInstanceActive(processInstance.getId(), ksession);
+            
+            countDownListener.waitTillCompleted();
+            logger.debug("First timer triggered");
+            countDownListener2.waitTillCompleted();
+            logger.debug("Second timer triggered");
+            countDownListener3.waitTillCompleted();
+            logger.debug("Third timer triggered");
+            countDownListener4.waitTillCompleted();
+            logger.debug("Fourth timer triggered");
+            
+            List<WorkItem> wi = handler.getWorkItems();
+            assertNotNull(wi);
+            assertEquals(3, wi.size());
+    
+            ksession.abortProcessInstance(processInstance.getId());
+        } finally {
+            System.clearProperty("jbpm.enable.multi.con");
```

Comment:
Just a minor thing -- but I believe we should reset the property to the original value instead of clearing it? (in case it would have been set to `false`, we would loose that setting).